api/v1alpha1/clustergroupupgrade_types.go

```diff
@@ -134,8 +134,12 @@ type ClusterGroupUpgradeStatus struct {
 	Conditions        []metav1.Condition `json:"conditions,omitempty"`
 	RemediationPlan   [][]string         `json:"remediationPlan,omitempty"`
 	ManagedPoliciesNs map[string]string  `json:"managedPoliciesNs,omitempty"`
-	Status            UpgradeStatus      `json:"status,omitempty"`
-	Precaching        PrecachingStatus   `json:"precaching,omitempty"`
+	// Contains the managed policies (and the namespaces) that have NonCompliant clusters
+	// that require updating.
+	ManagedPoliciesForUpgrade             []ManagedPolicyForUpgrade `json:"managedPoliciesForUpgrade,omitempty"`
+	ManagedPoliciesCompliantBeforeUpgrade []string                  `json:"managedPoliciesCompliantBeforeUpgrade,omitempty"`
```

Comment:
Looks like  ManagedPoliciesCompliantBeforeUpgrade  is just for record? In my opinion, I feel it's more clear to understand if we can keep one entry for managedPolicies and have all information there like, name, namespace and compliant status.

Reply:
Yes, I think it would be a good idea. I will add it to the list as changing it now would mean even more restructuring.